static/src/javascripts/projects/common/modules/crosswords/grid.js

```diff
@@ -4,172 +4,178 @@ import helpers from 'common/modules/crosswords/helpers';
-                return Grid.createHyphenSeparator(
-                    x,
-                    y,
-                    separatorDescription.direction
-                );
-            }
+const createSeparator = (
+    x: number,
+    y: number,
+    separatorDescription: ?SeparatorDescription
+): ?React.Element => {
+    if (separatorDescription) {
+        if (separatorDescription.separator === ',') {
+            return createWordSeparator(x, y, separatorDescription.direction);
+        } else if (separatorDescription.separator === '-') {
+            return createHyphenSeparator(x, y, separatorDescription.direction);
         }
     }
+};
 
-    getSeparators(x: number, y: number): ?Object {
-        return this.props.separators[helpers.clueMapKey(x, y)];
-    }
+export const Grid = (props: GridProps): React.Element<*> => {
+    const getSeparators = (x: number, y: number): ?SeparatorDescription =>
+        props.separators[helpers.clueMapKey(x, y)];
 
-    handleSelect(x: number, y: number): void {
-        this.props.crossword.onSelect(x, y);
-    }
+    const handleSelect = (x: number, y: number): void =>
+        props.crossword.onSelect(x, y);
+
+    const width = helpers.gridSize(props.columns);
+    const height = helpers.gridSize(props.rows);
+    const cells = [];
+    let separators = [];
+
+    const range = n => Array.from({ length: n }, (value, key) => key);
 
-    render(): Element {
-        const width = helpers.gridSize(this.props.columns);
-        const height = helpers.gridSize(this.props.rows);
-        const cells = [];
-        let separators = [];
-
-        const range = n => Array.from({ length: n }, (value, key) => key);
-
-        range(this.props.rows).forEach(y =>
-            range(this.props.columns).forEach(x => {
-                const cellProps = this.props.cells[x][y];
-
-                if (cellProps.isEditable) {
-                    const isHighlighted = this.props.crossword.isHighlighted(
-                        x,
-                        y
-                    );
-                    cells.push(
-                        React.createElement(
-                            Cell,
-                            Object.assign(
-                                {},
-                                cellProps,
-                                {
-                                    handleSelect: this.handleSelect,
-                                    x,
-                                    y,
-                                    key: `cell_${x}_${y}`,
-                                    isHighlighted,
-                                    isFocussed:
-                                        this.props.focussedCell &&
-                                        x === this.props.focussedCell.x &&
-                                        y === this.props.focussedCell.y,
-                                },
-                                this
-                            )
+    const cellsIn = props.cells;
```

Comment:
I got a linting error, saying incorrectly that `cells` wasn't used, if I didn't add this indirection.

Reply:
I guess this is a variation of the documented [Stateless Functional Component false positive](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md#false-positives-sfc), in which case, your workaround is the preferred solution

Reply:
Great detective work, that sounds exactly the issue. I'll add a comment.